test/integration/rememberable_test.rb

```diff
@@ -127,37 +127,83 @@ def cookie_expires(key)
     end
   end
 
-  test 'extends remember period when extend remember period config is true' do
+  test 'logging in sets remember_created_at if it is blank' do
     swap Devise, extend_remember_period: true, remember_for: 1.year do
-      create_user_and_remember
-      old_remember_token = nil
+      user = create_user
+      user.forget_me!
+
+      assert_nil user.remember_created_at
+
+      sign_in_as_user remember_me: true
+      user.reload
+
+      assert warden.user(:user) == user
+      assert_not_nil user.remember_created_at
+    end
+  end
+
+  test 'extends remember period on every authenticated request when extend remember period config is true' do
+    swap Devise, extend_remember_period: true, remember_for: 1.year, timeout_in: nil do
+      user = create_user_and_remember
+
+      get root_path
+      assert_response :success
 
-      travel_to 1.day.ago do
+      travel_to 1.day.from_now do
+        # tomorrow remember period is extended
         get root_path
-        old_remember_token = request.cookies['remember_user_token']
+        assert_response :success
       end
 
-      get root_path
-      current_remember_token = request.cookies['remember_user_token']
+      travel_to 6.months.from_now do
+        # 6 months later, still logged in
+        get root_path
+        assert_response :success
+      end
 
-      refute_equal old_remember_token, current_remember_token
+      travel_to 13.months.from_now do
+        # 13 months after remember_created_at was first set, we are still logged in because period was extended
+        get root_path
+        assert_response :success
+      end
+
+      travel_to 20.months.from_now do
+        # 20 months after remember_created_at was first set, we are still logged in because period was extended
+        get root_path
+        assert_response :success
+      end
+
+      travel_to 33.months.from_now do
+        # don't log in for over a year, we get logged out
```

Comment:
I'm pretty stuck getting this working, because I can't work out a way to get around warden's [short circuiting](https://github.com/wardencommunity/warden/blob/master/lib/warden/proxy.rb#L334).

Reply:
Shouldn't a timeout be set for these tests?

Otherwise the session itself has not timed out so the user is still logged in.

A valid remember token will prevent the session from timing out or automatically create one if the session doesn't exist. Neither a valid, invalid or expired remember token causes a logout.